src/icf/interfaceCallForwarder.ts

```diff
@@ -0,0 +1,463 @@
+import { compileCairo } from '../starknetCli';
+import * as path from 'path';
+import { logError } from '../utils/errors';
+import { SolcInterfaceGenOptions } from '../index';
+import fs from 'fs';
+import { AST } from '../ast/ast';
+
+import {
+  ArrayTypeName,
+  ASTContext,
+  ASTNode,
+  ASTNodeConstructor,
+  ASTNodeWriter,
+  ASTWriter,
+  ContractDefinition,
+  ContractKind,
+  DataLocation,
+  DefaultASTWriterMapping,
+  FunctionDefinition,
+  FunctionKind,
+  FunctionStateMutability,
+  FunctionVisibility,
+  Mutability,
+  ParameterList,
+  PragmaDirective,
+  PrettyFormatter,
+  SourceUnit,
+  StateVariableVisibility,
+  StructDefinition,
+  TypeName,
+  UserDefinedTypeName,
+  VariableDeclaration,
+} from 'solc-typed-ast';
+
+import { removeExcessNewlines } from '../utils/formatting';
+
+import {
+  createArrayTypeName,
+  createStaticArrayTypeName,
+  createUint256TypeName,
+} from '../utils/nodeTemplates';
+
+import { assert } from 'console';
+
+import { genCairoContract } from './genCairo';
+import { AbiType, FunctionAbiItemType, StructAbiItemType } from './abiTypes';
+import {
+  getAllStructsFromABI,
+  getFunctionItems,
+  getStructsFromABI,
+  transformType,
+  tupleParser,
+  typeToStructMapping,
+} from './utils';
+import { callClassHashScript, safeCanonicalHash } from '../export';
+
+const defaultSolcVersion = '0.8.14';
+
+/**
+ * Generate a cairo & solidity interface from given cairo contract file to interact
+ * with it (given cairo contract) from a solidity contract.
+ * More details you can find in the README.md file.
+ * @param filePath : path to the cairo contract file
+ * @param options : options for the generation (solc version, output directory, etc.)
+ * @returns : none
+ */
+export function generateSolInterface(filePath: string, options: SolcInterfaceGenOptions) {
+  const cairoPathRoot = filePath.slice(0, -'.cairo'.length);
+  const { success } = compileCairo(filePath, path.resolve(__dirname, '..'), {
+    debugInfo: false,
+  });
+
+  // files generated by the compilation of the given cairo contract
+  const resultPath = `${cairoPathRoot}_compiled.json`;
```

Comment:
You don't need to "calculate" these yourself, they are returned from compileCairo